src/Compilers/CSharp/Test/Syntax/Parsing/FuzzTesting.cs

```diff
@@ -81,18 +81,25 @@ public void c024928()
 
         [WorkItem(2771, "https://github.com/dotnet/roslyn/issues/2771")]
         [Fact]
-        public void TestBinary()
+        public unsafe void TestBinary()
         {
             // use a fixed seed so the test is reproducible
             var random = new System.Random(12345);
-            const int n = 40 * 1000 * 1000; // 40 million "character"s
-            var builder = new StringBuilder(n + 10);
-            for (int i = 0; i < n; i++)
+            // 40 million "character"s
+            const int n = 40 * 1000 * 1000; 
+            var chars = new char[n];
+            for (int i = 0; i < n - 1; i++)
             {
-                builder.Append((char)random.Next(0xffff));
+                chars[i] = (char) random.Next(char.MaxValue);
+            }
+            chars[n - 1] = (char) 0;
+            unsafe
+            {
+                fixed (char* bytePtr = chars)
+                {
+                    var tree = CSharpSyntaxTree.ParseText(new string(bytePtr));
+                }
```

Comment:
Good catch.  For whatever reason, I thought that the pointer constructor made a "view" over the byte ptr.

I'm going to use the unsafe string mutation method.